Refit/ApiResponse.cs

```diff
@@ -81,6 +81,6 @@ public interface IApiResponse : IDisposable
         HttpRequestMessage RequestMessage { get; }
         HttpStatusCode StatusCode { get; }
         Version Version { get; }
-        ApiException Error { get; }
+        Exception Error { get; }
```

Comment:
It is a breaking change.
Not sure it could be done.

Reply:
Good point. We could make it so that the custom exception has to derive from ApiException. That would allow people to have custom exception types. 

It's not clear to me what other exceptions people are looking to throw here?

Reply:
Ability to throw `UnauthorizedAccessException` was requested here:
https://github.com/reactiveui/refit/issues/562